tests/integrations/mcs/tso/proxy_test.go

```diff
@@ -128,42 +123,93 @@ func (s *tsoProxyTestSuite) TestTSOProxyWorksWithCancellation() {
 // TestTSOProxyStress tests the TSO Proxy can work correctly under the stress. gPRC and TSO failures are allowed,
 // but the TSO Proxy should not panic, blocked or deadlocked, and if it returns a timestamp, it should be a valid
 // timestamp monotonic increasing. After the stress, the TSO Proxy should still work correctly.
-func (s *tsoProxyTestSuite) TestTSOProxyStress() {
-	s.T().Skip("skip the stress test temporarily")
+func TestTSOProxyStress(t *testing.T) {
+	s := new(tsoProxyTestSuite)
+	s.SetT(&testing.T{})
+	s.SetupSuite()
 	re := s.Require()
-	// Add 1000 concurrent clients each round; 2 runs in total, and 2000 concurrent clients are created in total.
-	grpcClientConns := make([]*grpc.ClientConn, 0)
+
+	const (
+		totalRounds = 4
+		clientsIncr = 500
+		// The graceful period for TSO Proxy to recover from gPRC and TSO failures.
+		recoverySLA = 5 * time.Second
+	)
 	streams := make([]pdpb.PD_TsoClient, 0)
-	cancelFuncs := make([]context.CancelFunc, 0)
-	for i := 0; i < 2; i++ {
-		fmt.Printf("Start the %dth round of stress test with %d concurrent clients.\n", i, len(streams)+1000)
-		grpcClientConnsTemp, streamsTemp, cancelFuncsTemp := createTSOStreams(re, s.ctx, s.backendEndpoints, 1000, false)
-		grpcClientConns = append(grpcClientConns, grpcClientConnsTemp...)
+	cleanupFuncs := make([]testutil.CleanupFunc, 0)
+
+	// Start stress test for 90 seconds to avoid ci-test-job to timeout.
+	ctxTimeout, cancel := context.WithTimeout(s.ctx, 90*time.Second)
+	defer cancel()
+
+	// Push load from many concurrent clients in multiple rounds and increase the #client each round.
+	for i := 0; i < totalRounds; i++ {
+		log.Info("start a new round of stress test",
+			zap.Int("round-id", i), zap.Int("clients-count", len(streams)+clientsIncr))
+		streamsTemp, cleanupFuncsTemp :=
+			createTSOStreams(re, s.ctx, s.backendEndpoints, clientsIncr)
 		streams = append(streams, streamsTemp...)
-		cancelFuncs = append(cancelFuncs, cancelFuncsTemp...)
-		s.verifyTSOProxy(streams, 50, false)
+		cleanupFuncs = append(cleanupFuncs, cleanupFuncsTemp...)
+		s.verifyTSOProxy(ctxTimeout, streams, cleanupFuncs, 50, false)
+	}
+	s.cleanupGRPCStreams(cleanupFuncs)
+	log.Info("the stress test completed.")
+
+	// Wait for the TSO Proxy to recover from the stress.
+	time.Sleep(recoverySLA)
```

Comment:
Can we make it shorter?

Reply:
This is just test to make the test more stable.

Reply:
How about using something like eventually to make it as quick as possible?

Reply:
Ok, changed to 
	// Verify the TSO Proxy can still work correctly after the stress.
	testutil.Eventually(re, func() bool {
		err := s.verifyTSOProxy(s.ctx, s.streams, s.cleanupFuncs, 1, false)
		return err == nil
	}, testutil.WithWaitFor(recoverySLA), testutil.WithTickInterval(500*time.Millisecond))